### PR TITLE
Convert frameworks to be reactive

### PR DESCRIPTION
**Pull request by mraible**

### CODE REVIEW

**Comment by mraible** [comment]
@SentryMan When I try to run `java -jar target/helidon-se.jar`, I get an error:

```
Error: Unable to initialize main class com.okta.rest.HelloApplication
Caused by: java.lang.NoClassDefFoundError: io/helidon/security/spi/SecurityProvider
```

Any ideas?

Update: Strangely enough, it does work when I build it as a native image and run it.

File: helidon/pom.xml [hunk 1]
```diff
@@ -5,99 +5,38 @@
...
-        <plugins>
-            <plugin>
-                <groupId>org.apache.maven.plugins</groupId>
-                <artifactId>maven-dependency-plugin</artifactId>
-                <executions>
-                    <execution>
-                        <id>copy-libs</id>
-                    </execution>
-                </executions>
-            </plugin>
```
Comment (SentryMan) [inline helidon/pom.xml]: this should stay

**Review by SentryMan — CHANGES_REQUESTED** [review]
you're missing the copy dependency plugin

File: helidon/pom.xml (hunk 1 above)
Comment (mraible) [reply helidon/pom.xml]: That worked. Thanks!

File: README.md [hunk 2]
```diff
@@ -14,13 +14,13 @@ This repository contains example OAuth 2.0 resource servers built with Micronaut
 First, clone this repository:
 
 ```bash
-git clone https://github.com/oktadev/auth0-java-rest-api-examples.git
+git clone https://github.com/oktadev/auth0-java-rest-api-reactive-examples.git
```
Comment (mraible) [inline README.md]:
```suggestion
git clone https://github.com/oktadev/auth0-java-reactive-examples.git
```

File: demo.adoc [hunk 3]
```diff
@@ -3,9 +3,9 @@
 :toc: macro
 :source-highlighter: highlight.js
 
-=  Native Java REST API Demo Steps
+=  Native Reactive Java REST API Demo Steps
```
Comment (mraible) [inline demo.adoc]:
```suggestion
= Reactive Java REST API Demo Steps
```

File: README.md [hunk 4]
```diff
@@ -14,13 +14,13 @@ This repository contains example OAuth 2.0 resource servers built with Micronaut
 First, clone this repository:
```
Comment (mraible) [inline README.md]:
```suggestion
Click the button below to start a new GitPod development environment with this project:

[![Open in GitPod](https://gitpod.io/button/open-in-gitpod.svg)](https://gitpod.io/#https://github.com/oktadev/auth0-java-reactive-examples)

Or, if you'd rather try things locally, clone this repository:
```

File: demo.adoc [hunk 5]
```diff
@@ -39,7 +39,9 @@ auth0 test token -a https://<your-auth0-domain>/api/v2/ -s openid
 
   TOKEN=eyJraWQiOiJYa2pXdjMzTDRBYU1ZSzNGM...
 
-== Make a Java REST API with Micronaut
+== Make a Reactive Java REST API with Micronaut
+
+Nothing needs to be done to make Micronaut reactive. It has no separate reactive web framework. See Piotr Minkowski's https://piotrminkowski.com/2019/11/12/micronaut-tutorial-reactive/[Micronaut Tutorial: Reactive] and https://guides.micronaut.io/latest/tag-reactive.html[Micronaut's Reactive Guides] for more information.
```
Comment (mraible) [inline demo.adoc:44]: @sdelamo Can you please confirm this is truthful? 

Related: can you confirm that Micronaut enables virtual threads by default?

Comment (sdelamo) [reply demo.adoc:44]: If you use the Netty runtimes and you have a controller method which blocks (e.g. it calls a database with a blocking driver) then you should annotate your controller method with `@ExecuteOn(BLOCKING)`. If you use a Java version with supports virtual threads it uses virtual threads.

**Pull request merged**